mod-source-record-storage-server/src/main/java/org/folio/consumers/ParsedRecordChunksKafkaHandler.java

```diff
@@ -58,30 +60,30 @@ public ParsedRecordChunksKafkaHandler(@Autowired RecordService recordService,
   }
 
   @Override
-  public Future<String> handle(KafkaConsumerRecord<String, String> record) {
-    Event event = Json.decodeValue(record.value(), Event.class);
+  public Future<String> handle(KafkaConsumerRecord<String, String> targetRecord) {
+    Event event = Json.decodeValue(targetRecord.value(), Event.class);
     RecordCollection recordCollection = Json.decodeValue(event.getEventPayload(), RecordCollection.class);
 
-    List<KafkaHeader> kafkaHeaders = record.headers();
+    List<KafkaHeader> kafkaHeaders = targetRecord.headers();
 
     OkapiConnectionParams okapiConnectionParams = new OkapiConnectionParams(KafkaHeaderUtils.kafkaHeadersToMap(kafkaHeaders), vertx);
     String tenantId = okapiConnectionParams.getTenantId();
-    String correlationId = okapiConnectionParams.getHeaders().get("correlationId");
-    String key = record.key();
+    String chunkId = extractValueFromHeaders(targetRecord.headers(), CHUNK_ID_HEADER);
+    String key = targetRecord.key();
 
     int chunkNumber = chunkCounter.incrementAndGet();
 
     try {
-      LOGGER.debug("RecordCollection has been received, correlationId: {}, starting processing... chunkNumber {}-{}", correlationId, chunkNumber, key);
+      LOGGER.debug("RecordCollection has been received, chunkId: {}, starting processing... chunkNumber {}-{}", chunkId, chunkNumber, key);
```

Comment:
Is it possible to add an eventType here?

Reply:
Done